python/tvm/autotvm/measure/measure_methods.py

```diff
@@ -349,46 +349,49 @@ def _build_func_common(measure_input, check_gpu=None, cuda_arch=None, build_opti
     return func, tuple((get_const_tuple(x.shape), x.dtype) for x in args)
 
 
-def default_build_func(measure_input, tmp_dir, **kwargs):
-    """
-    Default build func. This can work for cuda, opencl, llvm backend
-
-    Parameters
-    ----------
-    measure_input: MeasureInput
-        The input of measurement
-    tmp_dir: str
-        The path of temporary directory to export generated library
-    """
-    tic = time.time()
-    try:
-        filename = os.path.join(tmp_dir, "tmp_func_%0x.tar" % getrandbits(64))
-        func, arg_info = _build_func_common(measure_input, **kwargs)
-        func.export_library(filename)
-    except Exception as e:  # pylint: disable=broad-except
-        return BuildResult(None, None, e, time.time() - tic)
-    return BuildResult(filename, arg_info, None, time.time() - tic)
-
-
-def android_ndk_build_func(measure_input, tmp_dir, **kwargs):
-    """
-    Build function for android device using ndk.
-
-    Parameters
-    ----------
-    measure_input: MeasureInput
-        The input of measurement
-    tmp_dir: str
-        The path of temporary directory to export generated library
-    """
-    tic = time.time()
-    try:
-        filename = os.path.join(tmp_dir, "tmp_func_%0x.so" % getrandbits(64))
-        func, arg_info = _build_func_common(measure_input, **kwargs)
-        func.export_library(filename, ndk.create_shared)
-    except Exception as e:  # pylint: disable=broad-except
-        return BuildResult(None, None, e, time.time() - tic)
-    return BuildResult(filename, arg_info, None, time.time() - tic)
+class DefaultBuilder:
```

Comment:
Since we already have `LocalBuilder` and `RPCBuilder`, which work on a different level, this name is very confusing.
We should remove this class and implement this as a function `create_build_func`
```python
def create_build_func(file_ext, fcompile, ...):
    def build_func_(...):
        ...
    return build_func
```

Reply:
Agree on the name issue. I renamed `DefaultBuilder` to `BuildFuncFactory`. I think that using Class should be fine here. It incapsulates `class params` and `build_func` method.

Reply:
replaced class `BuildFuncFactory` with function `make_build_func`